parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -174,6 +194,27 @@ public static ParquetValueReader<Record> recordReader(
     return new RecordReader(readers, struct);
   }
 
+  public static ParquetValueReader<?> replaceWithMetadataReader(
+      int id, ParquetValueReader<?> reader, Map<Integer, ?> idToConstant, int constantDL) {
+    if (id == MetadataColumns.ROW_ID.fieldId()) {
+      Long baseRowId = (Long) idToConstant.get(id);
+      return ParquetValueReaders.rowIds(baseRowId, reader);
+    } else if (id == MetadataColumns.LAST_UPDATED_SEQUENCE_NUMBER.fieldId()) {
+      Long baseRowId = (Long) idToConstant.get(id);
+      Long fileSeqNumber = (Long) idToConstant.get(id);
+      return ParquetValueReaders.lastUpdated(baseRowId, fileSeqNumber, reader);
```

Comment:
Does the last updated reader need to care about the baseRowId?

Reply:
Nvm saw your [comment](https://github.com/apache/iceberg/pull/12836/files#r2051043116) below, that rationale makes sense to me